src/GHC/Util/SrcLoc.hs

```diff
@@ -5,19 +5,48 @@ module GHC.Util.SrcLoc (
   , SrcSpanD(..)
   ) where
 
-import SrcLoc
-import Outputable
+import GHC.Types.SrcLoc
+import GHC.Utils.Outputable
 
 import Data.Default
 import Data.Data
 import Data.Generics.Uniplate.DataOnly
 
 -- 'stripLocs x' is 'x' with all contained source locs replaced by
 -- 'noSrcSpan'.
-stripLocs :: (Data from, HasSrcSpan from) => from -> from
+stripLocs :: Data from => from -> from
 stripLocs = transformBi (const noSrcSpan)
 
--- 'Duplicates.hs' requires 'SrcSpan' be in 'Default'.
+-- TODO (2020-10-03, SF): Maybe move the following definitions down to
+-- ghc-lib-parser at some point.
+
+-- 'Duplicates.hs' requires 'SrcSpan' be in 'Default' and 'Ord'.
 newtype SrcSpanD = SrcSpanD SrcSpan
-  deriving (Outputable, Eq, Ord)
+  deriving (Outputable, Eq)
 instance Default SrcSpanD where def = SrcSpanD noSrcSpan
+-- SrcSpan no longer provides 'Ord' so we are forced to roll our own.
+compareSrcSpans (SrcSpanD a) (SrcSpanD b) =
+  case a of
+    RealSrcSpan a1 _ ->
+      case b of
+        RealSrcSpan b1 _ ->
+          a1 `compareRealSrcSpans` b1
+        _ -> error "'Duplicate.hs' invariant error: can't compare unhelpful spans"
```

Comment:
This seems super dangerous. Could we say all unhelpful ones are equal? It screws up stuff, but in a less dangerous way.

Reply:
Yes we can.